generator/main.go

```diff
@@ -65,10 +74,30 @@ func recordWritesErrored(count float64) {
 	writesErrored.Add(count)
 }
 
+func recordPatchesCompleted(count float64) {
+	patchesCompleted.Add(count)
+}
+
+func recordPatchesErrored(count float64) {
+	patchesErrored.Add(count)
+}
+
+const defaultRocksetEndpoint = "https://api.rs2.usw2.rockset.com"
+
+var max_doc_id int
+
 func main() {
 	wps := mustGetEnvInt("WPS")
 	batchSize := mustGetEnvInt("BATCH_SIZE")
 	destination := mustGetEnvString("DESTINATION")
+	num_docs := getEnvDefaultInt("NUM_DOCS", -1)
+	do_patch := getEnvDefaultBool("PATCH", false)
```

Comment:
instead we can have a MODE here with values `ADD`, `PATCH` and in future `DELETE`. wdyt?

Reply:
Yes, I think that is much better